CONTRIBUTING.md

```diff
@@ -0,0 +1,248 @@
+## Resource Guidelines
+
+### Consider Pagination
+
+Most AWS resources are paginated and all resources should handle that.
+
+
+### Use Properties Instead of String Functions
+
+Currently, each resource can offter two functions to describe itself, that are
+used by the user to identify it and by *aws-nuke* to filter it.
+
+The String function is deprecated:
+
+```go
+String() string
+```
+
+The Properties function should be used instead:
+
+```go
+Properties() types.Properties
+```
+
+The interface for the String function is still there, because not all resources
+are migrated yet. Please use the Properties function for new resources.
+
+
+### Filter Resources That Cannot Get Removed
+
+Some AWS APIs list resources, that cannot be deleted. For example:
+
+* Resources that are already deleted, but still listed for some time (eg EC2 Instances).
+* Resources that are created by AWS, but cannot be deleted by the user (eg some IAM Roles).
+
+Those resources should be excluded in the filter step, rather than in the list step.
+
+
+## Styleguide
+
+### Go
+
+#### Code Format
+
+Like almost all Go projects, we are using `go fmt` as single source of truth
+for formatting the source code. Please use `go fmt` before committing any
+change.
+
+
+### Git
+
+#### Squash
+
+We want to keep the git history of *aws-nuke* clean. Generally, it is fine to
+have one commit per resource, but putting them into a single commit is also
+fine. But having a lot of meaningless "fixup" and "change" commits would
+clutter up the history. Therfore those commits should be squashed into a single
+commit.
+
+To squash all commits in your branch, execute:
+
+```
+git rebase --fork-point -i master
+```
+
+This opens an editor, where you should replace `pick` with `squash` on every
+but the first commit. Then *git* opens another editor, where you have to update
+the commit message. This message should be updated properly.
+
+Afterwards you have to force push your changes.
+
+> **Note:** It is generally not adviced to to a history rewrite, but we
+> consider that a branch and PR is owned by the author, until it gets merged.
+> Therefore the author can always rewrite its own branch. An important
+> implication is, that you should never add a commit to a branch of another
+> author, without communicating this beforehand.
+
+
+#### Rebase
+
+We want to keep the git history of *aws-nuke* clean. Using a merge from the
+master branch to update a feature branch would add unnecessary commits to the
+history. Therefore, please use rebase, rather than merge to update your branch.
+
+> **Note:** We cannot use the GitHub Squash Merge, since it would alter the
+> commit author. We do not want this, because it would not properly acknowledge
+> the contributions of the community.
+
+It is recommended to [squash](#squash) your branch before doing a rebase, so you avoid
+unnecessary conflicts.
+
+To rebase your branch, simply update your master branch and run this command:
+
+```
+git rebase master
```

Comment:
Shouldn't this be origin/master?

Reply:
No, if the `master` branch got updated, this will do just fine. Otherwise we would need to mention a `fetch`, too. Also the upstream might not be called origin, if someone forks our repo.